### PR TITLE
al2023 updates and some more

### PR DESCRIPTION
**Pull request by singholt**
### Summary

This PR updates the `ecs-logs-collector.sh` script with AL2023 related changes, and a few more misc updates.

### Implementation details

1. Collect system logs from systemd journal if `journalctl` is available.
2. Use token since its required for IMDSv2 (enabled by default on AL2023). Tokens are optional for IMDSv1. This is also applicable to other OS like AL2 where customer may enable IMDSv2.  We query IMDS to get the instance ID.
3. Collect logs from subfolders within `/var/log/ecs/` too, like `exec` and `service-connect` when available.
4. Remove '#' from commands in README for convenient copy and run.
5. Update copyright header and NOTICE.
6. Add AL2023 to the PR template.

### Testing

- [✔️] Works properly on Amazon Linux 2
- [✔️] Works properly on Amazon Linux 2023
- [✔️] Works properly on Ubuntu 20.04
- [✔️] Works properly on CentOS 8

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

**Review by YashdalfTheGray — APPROVED** [review]

File: ecs-logs-collector.sh [hunk 1]
```diff
@@ -279,7 +278,9 @@ try_set_instance_collectdir() {
 
   if test -z "$instance_id" && command -v curl > /dev/null; then
     info "getting instance id from ec2 metadata endpoint"
-    instance_id=$(curl --max-time 3 -s http://169.254.169.254/latest/meta-data/instance-id 2>/dev/null)
+    imds="http://169.254.169.254/latest"
+    token=$(curl -sS --retry 3 -X PUT "${imds}/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600" 2>/dev/null)
+    instance_id=$(curl -sS --retry 3 -H "X-aws-ec2-metadata-token: $token" "${imds}/meta-data/instance-id" 2>/dev/null)
```
Comment (chienhanlin) [inline ecs-logs-collector.sh:283]: Q. would this also work for the container instance with IMDSv1?

File: ecs-logs-collector.sh [hunk 2]
```diff
@@ -404,13 +405,19 @@ get_open_files() {
 
 get_common_logs() {
   try "collect common operating system logs"
+  mkdir -p "$info_system"
 
-  dstdir="${info_system}/var_log"
-  mkdir -p "$dstdir"
+  # On AL2023, system logs are stored in the systemd journal
+  if grep -q "Amazon Linux 2023" /etc/os-release; then
```
Comment (chienhanlin) [inline ecs-logs-collector.sh]: Q. would we have to update this line later for `Amazon Linux 202x`? or should we use wildcard here?

File: ecs-logs-collector.sh [hunk 3]
```diff
@@ -404,13 +405,19 @@ get_open_files() {
 
 get_common_logs() {
   try "collect common operating system logs"
```
Comment (chienhanlin) [inline ecs-logs-collector.sh:407]: If possible, could we have examples on this PR to show
1. A hierarchical structure of files/folders 
2. `system.log` collected from AL2023 will look like 
  
after making these changes?

Thanks!

File: ecs-logs-collector.sh (hunk 2 above)
Comment (singholt) [reply ecs-logs-collector.sh]: this is a good point. I changed it to check if `journalctl` is available, then we collect logs. works on all of AL2, AL2023, CentOS and Ubuntu!

File: ecs-logs-collector.sh (hunk 3 above)
Comment (singholt) [reply ecs-logs-collector.sh:407]: with the latest commit, the file structure remains the same as is today except a new `system.log` file will be added.

here's what the `system.log` file looks like (its very similar to `/var/log/ecs/messages`):
```
tail system.log 

Mar 24 16:12:53 ip-x-x-x-x.us-west-2.compute.internal dockerd[754716]: time="2023-03-24T16:12:53.666822696Z" level=debug msg=event module=libcontainerd namespace=moby topic=/tasks/exit
Mar 24 16:12:53 ip-x-x-x-x.us-west-2.compute.internal dockerd[754716]: time="2023-03-24T16:12:53.666904015Z" level=debug msg="attach: stderr: end"
Mar 24 16:12:53 ip-x-x-x-x.us-west-2.compute.internal dockerd[754716]: time="2023-03-24T16:12:53.666919408Z" level=debug msg="attach: stdout: end"
Mar 24 16:12:53 ip-x-x-x-x.us-west-2.compute.internal dockerd[754716]: time="2023-03-24T16:12:53.666929294Z" level=debug msg="attach done"
Mar 24 16:12:53 ip-x-x-x-x.us-west-2.compute.internal dockerd[754716]: time="2023-03-24T16:12:53.666951413Z" level=debug msg="Health check for container 2e70dbc94a8f25cac6feb04cd2f8bce022319602a621fc943fad40552e307e93 done (exitCode=0)"
Mar 24 16:12:54 ip-x-x-x-x.us-west-2.compute.internal systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Mar 24 16:12:54 ip-x-x-x-x.us-west-2.compute.internal audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Mar 24 16:12:54 ip-x-x-x-x.us-west-2.compute.internal audit: BPF prog-id=328 op=UNLOAD
Mar 24 16:12:54 ip-x-x-x-x.us-west-2.compute.internal audit: BPF prog-id=327 op=UNLOAD
Mar 24 16:12:54 ip-x-x-x-x.us-west-2.compute.internal audit: BPF prog-id=326 op=UNLOAD

```

Comment (chienhanlin) [reply ecs-logs-collector.sh:407]: Thanks!

**Review by chienhanlin — APPROVED** [review]

**Pull request merged**